### DISCUSSION TITLE
Debug Image Viewer pricing

### DISCUSSION THREAD

**Discussion by adrianboguszewski** [adrianboguszewski/debug-image-viewer#6] [General]
First, thanks a lot to all of you (77 people) who participated in the anonymous survey about the pricing model of the plugin's premium version. We appreciate the time you spent filling it and therefore we would like to answer some of your comments in this discussion.

The voting for the fair plugin price looks like the following:
0 USD/EUR - 15 people
<1 USD/EUR - 4 people
1 USD/EUR - 15 people
2 USD/EUR - 11 people
3 USD/EUR - 5 people
4 USD/EUR - 2 people
5 USD/EUR - 7 people
\>5 USD/EUR - 7 people

**Based on the voting we decided to reduce the price to 2EUR/month (individuals) with the release of version 4.2.0.**

1. **Many of you (15 people), but not the majority, suggested that the plugin should be just free** - well if the plugin were totally free, no new features would be implemented as the development would stop. I created the plugin because I needed it as a Computer Vision Engineer, but I'm no longer in that role... 
Just to remind you: the plugin is in the freemium model. It means all basic features (same features OpenCV offers in the imshow() function) are available for free and there are no plans to change this.
2. **Some of you advocated for a one-off payment for individuals and a subscription model for companies** - unfortunately, it's not possible in the JetBrains marketplace
3. **A few people said it's too much for just simple features like channel reverse** - we agree, it would be too much for just one feature, but we offer many more features for that fee (check all here: https://plugins.jetbrains.com/plugin/14371-debug-image-viewer-former-opencv-image-viewer-/features and https://plugins.jetbrains.com/plugin/14371-debug-image-viewer-former-opencv-image-viewer-/language-support)
4. **One person proposed 3- or 4-tier pricing model** - we would love to implement it, but it cannot be done in the current JetBrains environment
5. **The same person recommended pulling the first year fee way down and giving less discount in later years** - again, it's a great idea, but not doable now due to the marketplace limitations (there are rumors about that feature being WIP)
6. **One of you suggested focusing on donations as a revenue stream** - we did that before we implemented the subscription model. It just didn't work.
7. **Another person proposed to make the plugin free for students** - we vote for it, but we will do it when we finally have a big enough user base. All discounts for e.g. startups, non-profit organizations, and so on are already in place. It would be nice to apply discounts also for students (e.g. 75%), but JetBrains doesn't support any customization in that field.
8. **Someone also mentioned that normalization was a feature before and it's no longer the feature** - it has been moved to the premium version as stated in https://plugins.jetbrains.com/plugin/14371-debug-image-viewer-former-opencv-image-viewer-/freemium-transition

Many of your ideas are good and we agree with them, but they are now doable for now, as the JetBrains marketplace just supports very little customization.

Again, the free version has the same features as OpenCV imshow() offers. The premium version extends that support to many more libraries e.g. PyTorch, TensorFlow, Pillow + provides visualization helpers (https://plugins.jetbrains.com/plugin/14371-debug-image-viewer-former-opencv-image-viewer-/features)

**Comment by fenneishi** [comment]
Charging is reasonable, and it should be charged. I am an early user of this plugin, and the time it has saved me far exceeds the subscription fee.

**Reply by adrianboguszewski** [reply]
Thanks for that! I hope it helps you achieve your goals quicker 🙂
